tensorly/metrics/entropy.py

```diff
@@ -0,0 +1,34 @@
+from .. import backend as T
+from ..cp_tensor import cp_normalize
+import numpy as np
+
+# Author: Taylor Lee Patti <taylorpatti@g.harvard.edu>
+
+
+def vonNeumann_entropy(tensor, offset=1e-12):
+    """Returns the von Neumann entropy of the tensor.
+
+    Parameters
+    ----------
+    tensor : (matrix or CP tensor)
+        Data structure
+    offset : (float)
+        Minimum value at which to not consider eigenvalue zero to avoid singularities.
+
+    Returns
+    -------
+    float
+    """
+    if tensor.__class__.__name__ == 'CPTensor':
```

Comment:
```python3
if isinstance(tensor, CPTensor):
```

Reply:
Isn’t the entropy defined for a density matrix though? When will it be a higher-order tensor?

Reply:
We're going to sometimes organize it as a higher order tensor, at which point it will be more efficient to do the calculation in tensor form.

Reply:
```python3
    if isinstance(tensor, CPTensor):
        weights = cp_normalize(tensor).weights
        vn_entropy = -T.sum(weights * T.log2(weights))

    else:
        eig_vals = T.eigh(tensor)[0]
        eig_vals = eig_vals[eig_vals > offset]
        vn_entropy = -T.sum(T.log2(eig_vals) * eig_vals)

    return float(vn_entropy)
```

Reply:
How about:

    if isinstance(tensor, CPTensor):
        eig_vals = cp_normalize(tensor).weights

    else:
        eig_vals = T.eigh(tensor)[0]
        eig_vals = eig_vals[eig_vals > offset]

    vn_entropy = -T.sum(T.log2(eig_vals) * eig_vals)

    return float(vn_entropy)

Reply:
```python3
eig_vals = eig_vals[eig_vals > offset]
```
How about we keep this line outside the else statement?

Reply:
Sounds good. I'm also consolidating the last two lines.